src/ffi/mod.rs

```diff
@@ -831,6 +844,72 @@ extern "C" {
     fn fortify_source();
 }
 
+#[cfg(feature = "hide_ips")]
+struct LogCensor {
+    ip_mask_seed: [u8; 32],
+    ip_masks: parking_lot::Mutex<HashMap<Ipv4Addr, String>>,
+    ip_regex: Regex,
+}
+
+#[cfg(feature = "hide_ips")]
+impl LogCensor {
+    fn new() -> Option<Self> {
+        Regex::new(
+            r"(?x)\b
+                (?:
+                    (?:
+                        25[0-5]
+                        |  2[0-4][0-9]
+                        |  1[0-9]{2}
+                        |  [1-9]?[0-9]
+                    )
+                    \.
+                ){3}
+                (?:
+                    25[0-5]
+                    |  2[0-4][0-9]
+                    |  1[0-9]{2}
+                    |  [1-9]?[0-9]
+                )
+                \b",
```

Comment:
Aren't we using IPv6 only internally, so there would be no need to replace them?

Reply:
Hm, you might be right, but the task doesn't clarify this. I think it would be safer (and future proof) to also handle ipv6 but that is indeed more complexity. @Jauler - could you clarify this? :)

Reply:
I think it's the same precedence, also we might support ipv6 connections at some point, or if something is comming into tunnel, something like firewall, nat or any other component could capture ipv6 that are not controlled by us.

Reply:
Hmm, I'm not sure about this TBH - should I also replace all of the `::` with some hashes?